domains/MatchRoom.js

```diff
@@ -257,93 +267,132 @@ class MatchRoom extends Room {
 		return this.state.players.find(player => player.id === player_id);
 	}
 
-	onAdminMessage(message) {
-		const [command, ...args] = message;
-		let forward_to_views = true;
-		let update_admin = true;
+	getMaxPossiblePlayers() {
+		return Math.min(this.getViewMeta().players || Infinity, MAX_PLAYERS);
+	}
 
-		// TODO: Extract this encompassing try..catch to own method
-		try {
-			switch (command) {
-				case 'getState': {
-					forward_to_views = false;
-					break;
-				}
+	initAutoJoin() {
+		// 1. drop all players
```

Comment:
I would not drop previous players here, just insert the other ones into empty slots. I do not see any upside to revoke manual settings just to sort all slots by arrival. I would prioritize keeping assigned slots a stable as possible.

One implementation would be after sorting to remove all players which are assigned to slots from the sorted list and then call autoJoinUser(user) for the remaining ones.

Reply:
Thanks, I've implemented the suggestion!